pkg/scyllaclient/client_rclone.go

```diff
@@ -460,22 +462,43 @@ func (c *Client) RcloneListDirIter(ctx context.Context, host, remotePath string,
 		}
 	}
 
-	var v RcloneListDirItem
-	for dec.More() {
-		// Detect context cancellation
-		if ctx.Err() != nil {
-			return ctx.Err()
+	errCh := make(chan error)
+	go func(ch chan error) {
+		var v RcloneListDirItem
+		for dec.More() {
+			// Detect context cancellation
+			if ctx.Err() != nil {
+				ch <- ctx.Err()
+				break
+			}
+
+			// Read value
+			v = RcloneListDirItem{}
+			if err := dec.Decode(&v); err != nil {
+				ch <- err
+			}
+			f(&v)
+
+			ch <- nil
```

Comment:
It would be even nicer to have just `ch <- err` sometimes the err is nil

Reply:
Left this as-is because in both cases where err gets set we now return immediately.